src/compo/viirs_aod2ioda.py

```diff
@@ -107,7 +184,70 @@ def _read(self):
 
             qcall = ncd.variables['QCAll'][:].ravel().astype('int32')
             obs_time = np.full(np.shape(qcall), base_datetime, dtype=object)
-            if self.mask == "maskout":
+
+            ncd.close()
+
+            # apply ADP mask if indicated by user
+            if self.adp_mask is not None:
+
+                # Check if ADP file start/end times match AOD file (creation time can differ slightly)
+                # Using negative indexing makes the paths irrelevant for the comparison
+                if self.adp_mask[n][-53:-20] != f[-53:-20]:
+                    print(f'ADP data file {self.adp_mask[n]} start and end times do not match those of AOD file {f}')
+                    print(f'Continuing to next AOD and ADP file (or ending)')
+                    continue
+
+                print('Processing the following AOD and ADP files:')
+                print(f)
+                print(self.adp_mask[n])
+                print()
```

Comment:
make sure you remove unnecessary print statements

Reply:
Thanks Jerome, I was wondering if I would need to create a test for this. Also, sorry for the slow reply, I was out on vacation last week.

Regarding the test, would it be preferred for me to expand the current test that exists for the IODA converter, or should I add a new, separate test to cover the new functionality? 

Regarding the print statements, I had added those just before opening the PR as I thought it may be good for users to have a log of what files were processed, but I've got no problem removing those if that is unneccessary or unwanted.